### PR TITLE
[RLlib] RLlib deprecation Notices Part 2 (models/tf, models/torch, base_mode, catalog, modelv2, models/temp_spec_classes, policy/)

### PR DESCRIPTION
**Pull request by avnishn**
Signed-off-by: Avnish <avnishnarayan@gmail.com>

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/models/tf/noop.py [hunk 1]
```diff
@@ -13,4 +15,6 @@ class NoopModel(TFModelV2):
 
     @override(ModelV2)
     def forward(self, input_dict, state, seq_lens):
+        if log_once("rllib_tf_noop_model_deprecation"):
```
Comment (avnishn) [inline rllib/models/tf/noop.py:18]: need to use decorator instead.

File: rllib/models/tf/primitives.py [hunk 2]
```diff
@@ -52,4 +56,6 @@ class IdentityNetwork(tf.keras.Model):
 
     @override(tf.keras.Model)
     def call(self, inputs, training=None, mask=None):
+        if log_once("rllib_models_tf_IdentityNetwork_deprecation"):
```
Comment (avnishn) [inline rllib/models/tf/primitives.py:59]: use decorator instead.

File: rllib/models/tf/recurrent_net.py [hunk 3]
```diff
@@ -67,6 +69,10 @@ def forward(
         """Adds time dimension to batch before sending inputs to forward_rnn().
 
         You should implement forward_rnn() in your subclass."""
+        if log_once("rllib_tf_rnn_model_deprecation"):
```
Comment (avnishn) [inline rllib/models/tf/recurrent_net.py:72]: use decorator instead.

File: rllib/models/torch/mingpt.py [hunk 4]
```diff
@@ -48,6 +50,8 @@ class NewGELU(nn.Module):
     """
 
     def forward(self, x):
+        if log_once("torch_gelu_deprecation"):
```
Comment (avnishn) [inline rllib/models/torch/mingpt.py:53]: add Deprecation decorator.

File: rllib/models/torch/model.py [hunk 5]
```diff
@@ -27,6 +29,8 @@ def save(self, path: str) -> None:
             path: Path on disk the checkpoint is saved to
 
         """
+        if log_once("torch_model_io_deprecation"):
```
Comment (avnishn) [inline rllib/models/torch/model.py:32]: add deprecation decorator.

File: rllib/models/torch/noop.py [hunk 6]
```diff
@@ -10,4 +12,6 @@ class TorchNoopModel(TorchModelV2):
 
     @override(ModelV2)
     def forward(self, input_dict, state, seq_lens):
+        if log_once("torch_noop_model_deprecation"):
```
Comment (avnishn) [inline rllib/models/torch/noop.py:15]: add deprecation decorator.

File: rllib/models/torch/recurrent_net.py [hunk 7]
```diff
@@ -74,6 +76,10 @@ def forward(
         """Adds time dimension to batch before sending inputs to forward_rnn().
 
         You should implement forward_rnn() in your subclass."""
+        if log_once("torch_rnn_network_deprecaiton"):
```
Comment (avnishn) [inline rllib/models/torch/recurrent_net.py:79]: add deprecation decorator.

File: rllib/models/torch/torch_action_dist.py [hunk 8]
```diff
@@ -457,6 +507,14 @@ class TorchDeterministic(TorchDistributionWrapper):
     requiring the "mean" values as NN output).
     """
 
+    def __init__(*args, **kwargs):
+        if log_once("torch_action_dist_deterministic_deprecation"):
```
Comment (avnishn) [inline rllib/models/torch/torch_action_dist.py:511]: add decorator instead.

**Review by avnishn — COMMENTED** [review]
first pass review

**Comment by Rohan138** [comment]
lgtm, not a codeowner, @kouroshHakha can you stamp?

**Comment by kouroshHakha** [comment]
@avnishn tests are failing.

**Review by kouroshHakha — APPROVED** [review]

**Pull request merged**